src/rtc_peer.ts

```diff
@@ -269,6 +329,16 @@ export class RTCPeer extends EventEmitter {
             if (this.candidates.length > 0) {
                 this.flushICECandidates();
             }
+
+            if (!this.dcNegotiated) {
+                this.dcNegotiated = true;
+            } else if (this.dc.readyState === 'open') {
+                this.logger.logDebug('RTCPeer.signal: unlocking signaling lock');
+                this.dc.send(encodeDCMsg(this.enc, DCMessageType.Unlock));
+            } else {
+                this.logger.logWarn('RTCPeer.signal: dc not open upon receiving answer');
+            }
```

Comment:
Can you explain this a bit -- I think I missed something. Is it possible to get to the else?

Reply:
@cpoile I think so, if the data channel is closed by the time we get an answer. Calling `dc.send` on a non-open channel will throw, so we are mostly guarding against that.